custom_connectors/custom_auth/trackvia_connector.rb

```diff
@@ -0,0 +1,787 @@
+          control_type: call(:get_control_type,
+                             type: field['type']),
+          properties: call(:get_properties,
+                           type: field['type'])
+        }
+      end
+    end,
+
+    get_fields: lambda do |input|
+      view_id = input[:view_id]
+      structure = get("/openapi/views/#{view_id}")['structure']
+                  .reject { |field| !field['canCreate'] || !field['canUpdate'] }
+      structure.map do |field|
+        {
+          name: field['name'],
+          label: field['name'],
+          optional: !field['required'],
+          type: call(:get_type, type: field['type']),
+          control_type: call(:get_control_type,
+                             type: field['type'],
+                             name: field['name']),
+          pick_list: call(:get_picklist_options,
+                          choices: field['choices']),
+          delimiter: call(:get_delimeter,
+                          type: field['type']),
+          properties: call(:get_properties,
+                           type: field['type']),
+          toggle_hint: call(:get_toggle_hint,
+                            type: field['type'],
+                            name: field['name']),
+          toggle_field: call(:get_toggle_field,
+                             type: field['type'],
+                             name: field['name'],
+                             required: !field['required'])
+        }
+      end
+    end,
+
+    get_fields_sample_output: lambda do |input|
+      view_id = input[:view_id]
+      get("/openapi/views/#{view_id}?start=0&max=1")
+        .dig('data', 0)
+    end
+  },
+  object_definitions: {
+    app: {
+      fields: lambda do
+        [
+          { name: 'name' },
+          { name: 'id', type: :integer }
+        ]
+      end
+    },
+    view: {
+      fields: lambda do
+        [
+          { name: 'id', type: :integer },
+          { name: 'name' },
+          { name: 'applicationName' },
+          { name: 'default', type: :boolean }
+        ]
+      end
+    },
+    record: {
+      fields: lambda { |_connection, config_fields|
+        call(:get_fields, view_id: config_fields['view_id'])
+      }
+    },
+    response_record: {
+      fields: lambda { |_connection, config_fields|
+        call(:get_output_fields, view_id: config_fields['view_id'])
+      }
+    },
+    hook_body: {
+      fields: lambda { |_connection, config_fields|
+        call(:get_output_fields, view_id: config_fields['view_id'])
+      }
+    },
+    request: {
+      fields: lambda do |_connection, config_fields|
+        [
+          {
+            name: 'data',
+            type: :array,
+            of: :object,
+            properties: [
+              call(:get_fields, view_id: config_fields['view_id'])
+            ]
+          }
+        ]
+      end
+    },
+    user: {
+      fields: lambda do
+        [
+          { name: 'Status' },
+          { name: 'Time Zone' },
```

Comment:
@svchandra I'm not entirely sure what problem this is attempting to resolve... I have been using both data pills and fields with gaps with no issue. The only time that I've seen issues with spaces is when I attempted inject a value in the URL that contained a string. I've since fixed that issue. Almost all of TrackVia's API URLs will exclusively take numbers as the url parameters. Can you provide a concrete example of how one might run into an issue here?

Reply:
@skennicutt Please remove un-used object definitions, this will address my comment.
e.g.
-  view
- user
- column